modules/svg/data.js

```diff
@@ -500,29 +500,41 @@ export function svgData(projection, context, dispatch) {
         return _src || '';
     };
 
+    function flattenGeoms(c, type) {
+        /* eslint-disable no-fallthrough */
+        switch (type) {
+            case 'Point':
+                c = [c];
+            case 'MultiPoint':
+            case 'LineString':
+                break;
+
+            case 'GeometryCollection':
+            case 'MultiPolygon':
+                c = _flatten(c);
+            case 'Polygon':
+            case 'MultiLineString':
+                c = _flatten(c);
+                break;
+        }
+        /* eslint-enable no-fallthrough */
+
+        return c;
+    }
+
     function flattenCoords() {
         var features = getFeatures(_geojson);
         if (!features.length) return;
 
         var coords = _reduce(features, function(coords, feature) {
-            var c = feature.geometry.coordinates;
-
-            /* eslint-disable no-fallthrough */
-            switch (feature.geometry.type) {
-                case 'Point':
-                    c = [c];
-                case 'MultiPoint':
-                case 'LineString':
-                    break;
-
-                case 'MultiPolygon':
-                    c = _flatten(c);
-                case 'Polygon':
-                case 'MultiLineString':
-                    c = _flatten(c);
-                    break;
+            var c = [];
+            if (feature.geometry.geometries) {
+                feature.geometry.geometries.forEach( function(geom) {
+                    c = c.concat(flattenGeoms(geom.coordinates, geom.type));
+                });
+            } else {
+                c = flattenGeoms(feature.geometry.coordinates, feature.geometry.type);
```

Comment:
really good approach, better than turning geometry collection into feature array in the geojson